### ISSUE TITLE
[RFC] Node.prototype.process

### ISSUE THREAD

**Issue by aleclarson** [aleclarson/nebu#4]
The `process` method would apply a new set of plugins to the node and its descendants.

This may require changes to `magic-string` to work properly. We'll want a "slice" of the `MagicString` object that applies changes based on a character offset (where our node starts). The original `MagicString` object would inherit any changes we make to the "slice".

```js
// A plugin that applies its own plugins.
const plugin = {
  ClassBody(node) {
    node.process(plugins, state)
  }
}
```

The `plugins` argument must be an array with at least one element.

The `state` argument is an optional object, which defaults to the "parent" state.

This is very efficient when you want a subset of plugins to be used only if within a certain context.

**Comment by aleclarson** [comment]
This has been added in 4015b39d86fe75b90fe3e68dc603d82f21361083 (not available on NPM yet).

```js
node.process(state, plugins)
// or
node.process(plugins)
```

I've decided the `state` argument **won't** default to the "parent" state (that is, the `state` of either `nebu.process` or another `node.process` call that we're nested in). It's easy enough to pass in the parent state manually, since it's already exposed to every visitor.
